### PR TITLE
Add Learning Objectives

### PR DESCRIPTION
**Pull request by brymz**
(As per #211)

When adding the learning objectives list to the assignment markdown, the ordered list generated by assignment.html is indented. I think it makes the page a little lopsided, but I couldn't figure out why it was doing that or how to fix it. Any ideas? 

Are there other issues with the formatting? I figure it's best to settle formatting on the first one.

### CODE REVIEW

**Comment by ethanwhite** [comment]
I don't find the list formatting too distracting.

The `Learning Objectives` and `Exercises` headers should be proper headers. So `### Exercises` instead of `***- Exercises***`. I'd drop the leading dash.

**Comment by brymz** [comment]
> headers should be proper headers. 

Done. I've also added learning objectives to the remaining assignments and included them in the PR.

File: assignments/git-1-R.md [hunk 1]
```diff
@@ -5,4 +5,15 @@ language: R
 exercises: ['Version Control 1', 'Version Control 2', 'Version Control 3', 'Version Control 4', 'Version Control 5', 'Version Control 6']
 ---
 
+####Learning Objectives
+
+> Following this assignment students should be able to:
+
+> - establish a GitHub local repository
```
Comment (ethanwhite) [inline assignments/git-1-R.md]: If it's local it isn't GitHub. Regardless I like keeping this big picture so "establish a local version control repository"

Comment (ethanwhite) [inline assignments/git-1-R.md]: If it's local it isn't GitHub. Regardless I like keeping this big picture so "establish a local version control repository"

File: assignments/r-10.md [hunk 2]
```diff
@@ -6,4 +6,14 @@ language: R
 exercises: ['Higher Order Functions', 'Regular Expressions', 'Debugging', 'Tests']
 ---
 
+
+####Learning Objectives
+
+> Following this assignment students should be able to:
+
+> - write basic and linked functions
```
Comment (ethanwhite) [inline assignments/r-10.md]: What is a "linked function"?

File: assignments/r-2.md [hunk 3]
```diff
@@ -3,7 +3,18 @@ layout: page
...
 
+####Learning Objectives
+
+> Following this assignment students should be able to:
+
+> - understand the vector and data frame object structures
+> - assign, subset, and manipulate data in a vector
+> - execute vector algebra
+> - import .CSV data files as a data frame
+
```
Comment (ethanwhite) [inline assignments/r-2.md]: Change the last bullet to "create data frames"

Add one more bullet:

* subset data from a data frame

File: assignments/r-4.md [hunk 4]
```diff
@@ -6,4 +6,14 @@ language: R
 exercises: ['Graphing 1', 'Graphing 2', 'Statistics 1']
 ---
 
+####Learning Objectives
+
+> Following this assignment students should be able to:
+
+> - understand the basic plot function of `ggplot2`
+> - import 'real' data with missing values and extra lines
```
Comment (ethanwhite) [inline assignments/r-4.md]: instead of 'real' what about 'messy' or 'realistic'?

File: assignments/r-6.md [hunk 5]
```diff
@@ -6,4 +6,13 @@ language: R
 exercises: ['Making Choices 1', 'Making Choices 2', 'Strings 8', 'Making Choices 3']
 ---
 
+####Learning Objectives
+
+> Following this assignment students should be able to:
+
+> - understand and use the basic choice operators
```
Comment (ethanwhite) [inline assignments/r-6.md]: I'm guessing by "choice operators" you mean `==`, `>`, `<=`, etc., in which case the formal term (which we should use) is "relational operators".

File: assignments/sql-1.md [hunk 6]
```diff
@@ -5,4 +5,15 @@ language: SQL
 exercises: ['QAQC 1', 'Importing Data', 'Creating Tables', 'Adding Records', 'Updating Records', 'Basic Queries', 'Filtering', 'Sorting', 'Distinct', 'Missing Data', 'Grouping', 'Aggregation 1', 'Aggregation 2']
 ---
 
+####Learning Objectives
+
+> Following this assignment students should be able to:
+
+> - set-up an Excel spreadsheet for quality control at data entry
```
Comment (ethanwhite) [inline assignments/sql-1.md]: how about a little broader: "implement quality control for data entry in spreadsheets"

File: assignments/sql-1.md [hunk 7]
```diff
@@ -5,4 +5,15 @@ language: SQL
...
+> - import existing data and create new data using SQLite
+> - understand the basic query structure of SQL
+> - execute SQL commands in SQLite to select, sort, group, and aggregate data
```
Comment (ethanwhite) [inline assignments/sql-1.md]: drop the SQLite here for generality

File: assignments/sql-1.md [hunk 8]
```diff
@@ -5,4 +5,15 @@ language: SQL
 exercises: ['QAQC 1', 'Importing Data', 'Creating Tables', 'Adding Records', 'Updating Records', 'Basic Queries', 'Filtering', 'Sorting', 'Distinct', 'Missing Data', 'Grouping', 'Aggregation 1', 'Aggregation 2']
 ---
 
+####Learning Objectives
+
+> Following this assignment students should be able to:
+
+> - set-up an Excel spreadsheet for quality control at data entry
+> - import existing data and create new data using SQLite
```
Comment (ethanwhite) [inline assignments/sql-1.md]: how about: "import and create data in an SQL database"

File: assignments/sql-2.md [hunk 9]
```diff
@@ -5,4 +5,14 @@ language: SQL
 exercises: ['Joins 1', 'Joins 2', 'Joins 3', 'Joins 4', 'Nested Queries', 'Tidy Data',  'Database Structure 1', 'Database Structure 2']
 ---
 
+####Learning Objectives
+
+> Following this assignment students should be able to:
+
+> - execute SQL commands in SQLite to join tables
```
Comment (ethanwhite) [inline assignments/sql-2.md]: "Use joins to combine tables in SQL"

**Comment by ethanwhite** [comment]
I've added minor in-line comments throughout. Otherwise looks great!

File: assignments/git-1-R.md (hunk 1 above)
Comment (brymz) [inline assignments/git-1-R.md]: Done.

File: assignments/r-10.md (hunk 2 above)
Comment (brymz) [inline assignments/r-10.md]: I'm not sure to be honest. I thought I got the phrase from a previous version of the course. It's the subtitle to Functions-4-R.md. But if it wasn't from you... I think we're trying to say 'nested' functions. Though I am also unclear if there is a difference between 'nested' and 'higher order' functions.

I've revised to 'nested functions'.

File: assignments/r-2.md (hunk 3 above)
Comment (brymz) [inline assignments/r-2.md]: Done.

File: assignments/r-4.md (hunk 4 above)
Comment (brymz) [inline assignments/r-4.md]: I went with 'messy'.

File: assignments/r-6.md (hunk 5 above)
Comment (brymz) [inline assignments/r-6.md]: Done.

File: assignments/sql-1.md (hunk 6 above)
Comment (brymz) [inline assignments/sql-1.md]: Done.

File: assignments/sql-1.md (hunk 7 above)
Comment (brymz) [inline assignments/sql-1.md]: Remove throughout. Use alternate text in Objective 2: 'new data in a database'.

File: assignments/sql-1.md (hunk 8 above)
Comment (brymz) [inline assignments/sql-1.md]: Done.

**Comment by brymz** [comment]
Revised as per the line comments above. Also, the subtitle of Functions-4-R.md is modified from 'Linking Functions' to 'Nested Functions'.

**Comment by ethanwhite** [comment]
Thanks!

**Pull request merged**